pkg/tfbridge/serve.go

```diff
@@ -26,19 +26,14 @@ import (
 
 // Serve fires up a Pulumi resource provider listening to inbound gRPC traffic,
 // and translates calls from Pulumi into actions against the provided Terraform Provider.
-func Serve(module string, version string, info ProviderInfo, pulumiSchema []byte, options ...Option) error {
-	opts := opts{}
-	for _, applyOption := range options {
-		applyOption(&opts)
-	}
-
+func Serve(module string, version string, info ProviderInfo, pulumiSchema []byte) error {
 	// Create a new resource provider server and listen for and serve incoming connections.
 	return provider.Main(module, func(host *provider.HostClient) (pulumirpc.ResourceProviderServer, error) {
-		if len(opts.muxWith) > 0 {
+		if info.MuxWith != nil && len(info.MuxWith) > 0 {
```

Comment:
We should probably encapsulate this whole block L32-L57 in its own function: `newMuxWithProvider` to make testing easier. We can then test the runtime behavior similar to this:

https://github.com/pulumi/pulumi-terraform-bridge/blob/2bb619413b2cca3d6acec58c76acb25dbb844fcb/pkg/tests/regress_923_test.go#L29

Reply:
@iwahbe First: What signature is required for the new method `newMuxWithProvider`? Currently the code relies completely on closures. They must be unwound using parameters of the new function. From my point of view we need a combination of `Serve` and `provider.Main`. Not sure if being able to test this satisfies the effort. The parameter list might be get long and ugly.

Second: how/when will the method being called? After `if len(opts.muxWith) > 0 {` or in some other way?

Reply:
I played around with this vis a vis testing. I came up with [this](https://github.com/tmeckel/pulumi-terraform-bridge/pull/1).

Reply:
@iwahbe merged

Reply:
I've to see with what I can come up with.

Reply:
@t0yv0 @iwahbe initial muxing tests implemented